patches/core-sdk/0003-Enable-building-on-arm64-machines.patch

```diff
@@ -0,0 +1,116 @@
+From 4b5c617203cfb9d2c1b12995e12d819fba6d7b6f Mon Sep 17 00:00:00 2001
+From: Omair Majid <omajid@redhat.com>
+Date: Tue, 8 Oct 2019 17:02:29 -0400
+Subject: [PATCH] Enable building on arm64 machines
+
+With this commit, I can build core-sdk on RHEL 8 on arm64 directly,
+without cross compilation.
+
+Bump the sourcelink version to pick up the ability to parse git info
+without depending on libgit2sharp. This allows sourcelink to work on
+arm64. The version is the same as the one recently added to core-setup:
+https://github.com/dotnet/core-setup/pull/7696
+
+Introduce a new 'BuildArchitecture' msbuild property that contains the host
+architecture (arm64, x64, etc). This is the architecture of the
+currently running machine, and may be different from the architecture we
+are targetting in the case of cross compilation.
+
+There's a gotcha with BuildArchitecture: under Visual Studio (an x86) process,
+we generally want a x64 architecture. So try and restrict it to arm64 only.
+
+Use BuildArchitecture to determine whether _crossDir and LibCLRJitRid need to
+be special-cased for arm64 or or not.
+---
+ Directory.Build.props                            | 6 ++++++
+ eng/Versions.props                               | 2 +-
+ src/redist/targets/Crossgen.targets              | 6 +++---
+ src/redist/targets/GenerateLayout.targets        | 4 ++--
+ src/redist/targets/GetRuntimeInformation.targets | 1 -
+ 5 files changed, 12 insertions(+), 7 deletions(-)
+
+diff --git a/Directory.Build.props b/Directory.Build.props
+index b65a72410..be3834859 100644
+--- a/Directory.Build.props
++++ b/Directory.Build.props
+@@ -7,6 +7,12 @@
+     <PackageLicenseExpression>MIT</PackageLicenseExpression>
+   </PropertyGroup>
+ 
++  <PropertyGroup>
++    <BuildArchitecture>$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture.ToString().ToLowerInvariant())</BuildArchitecture>
++    <Architecture Condition="'$(BuildArchitecture)' == 'arm64'">$(BuildArchitecture)</Architecture>
++    <Architecture Condition="'$(Architecture)' == ''">x64</Architecture>
++  </PropertyGroup>
++
+   <PropertyGroup>
+     <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
+     <DebugType>embedded</DebugType>
+diff --git a/eng/Versions.props b/eng/Versions.props
+index 344a325bf..37e478e72 100644
+--- a/eng/Versions.props
++++ b/eng/Versions.props
+@@ -87,7 +87,7 @@
+     <VersionToolsVersion>$(BuildTasksFeedToolVersion)</VersionToolsVersion>
+     <DotnetDebToolVersion>2.0.0</DotnetDebToolVersion>
+     <MicrosoftNETTestSdkVersion>15.8.0</MicrosoftNETTestSdkVersion>
+-    <MicrosoftSourceLinkVersion>1.0.0-beta2-18618-05</MicrosoftSourceLinkVersion>
++    <MicrosoftSourceLinkVersion>1.0.0-beta2-19367-01</MicrosoftSourceLinkVersion>
```

Comment:
This is interesting - this version should be getting overridden by our built SourceLink version anyway.  I'll take a look at that.

Reply:
I added the entire `git format-patch` from the core-sdk PR: https://github.com/dotnet/core-sdk/pull/4102/. It could very well be that source-build doesn't care about this property. Do you want me to test out removing this? Or leave as it is, like the change that was merged into core-sdk?

Btw, the same applies for all the new patches added b this PR. They are the `git format-patch` output from the individual repos, tweaked to apply to source-build's `release/3.1`, but without any other changes.

Reply:
I checked and the override is working in my local build, so we must have a new enough version of SourceLink in source-build if this is working for you.  I updated it not too long ago so that makes sense.  I'd prefer leaving this change out if we can just because version properties like this tend to conflict quite a bit.

Reply:
Okay. I am sold!